### PR TITLE
Introduce a general cache tag invalidator service

### PR DESCRIPTION
**Pull request by Toflar**
Our cache invalidation logic currently operates directly on the HTTP cache only which sometimes is not sufficient because when a user e.g. invalidates a content element, you may also want to invalidate some API cache information or whatever.

This has to be done on Contao level and not in a general HttpCache(Bundle) event because it's logically incorrect there. The HttpCache(Bundle) operates on the HTTP cache. What we want here is a general tag invalidation. Invalidating the HTTP cache tags is **one** of the steps but thanks to this new event, it can be many more.

### CODE REVIEW

File: core-bundle/src/Event/InvalidateCacheTagsEvent.php [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Event;
+
+class InvalidateCacheTagsEvent
+{
+    /**
+     * @param array<string> $tags
```
Comment (m-vo) [inline core-bundle/src/Event/InvalidateCacheTagsEvent.php]:
```suggestion
     * @param list<string> $tags
```

**Review by m-vo — DISMISSED** [review]

**Review by leofeyer — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @Toflar.

**Comment by aschempp** [comment]
Wouldn't it make more sense to add this one missing method to `EntityCacheTags` and maybe rename that class to not be related to entities only? Like have one central service to add and invalidate cache tags from strings and anything else?

**Comment by Toflar** [comment]
I don't understand a single word. Maybe just propose your changes? This is merged already anyway (and also used in core).

**Comment by aschempp** [comment]
see https://github.com/contao/contao/pull/7780

**Pull request merged**